umbra-js/src/classes/Umbra.ts

```diff
@@ -175,55 +154,123 @@ export class Umbra {
     let tx: ContractTransaction;
     if (isETH(token)) {
       const txOverrides = { ...filteredOverrides, value: toll.add(amount) };
-      tx = await this.umbraContract.sendEth(
-        stealthKeyPair.address,
-        compressedXCoordinate,
-        encrypted.ciphertext,
-        txOverrides
-      );
+      tx = await this.umbraContract
+        .connect(txSigner)
+        .sendEth(
+          stealthKeyPair.address,
+          toll,
+          compressedXCoordinate,
+          encrypted.ciphertext,
+          txOverrides
+        );
     } else {
       const txOverrides = { ...filteredOverrides, value: toll };
-      tx = await this.umbraContract.sendToken(
-        stealthKeyPair.address,
-        token,
-        amount,
-        compressedXCoordinate,
-        encrypted.ciphertext,
-        txOverrides
-      );
+      tx = await this.umbraContract
+        .connect(txSigner)
+        .sendToken(
+          stealthKeyPair.address,
+          token,
+          amount,
+          compressedXCoordinate,
+          encrypted.ciphertext,
+          txOverrides
+        );
     }
 
     // We do not wait for the transaction to be mined before returning it
     return { tx, stealthKeyPair };
   }
 
   /**
-   * @notice Withdraw funds for a recipient
+   * @notice Withdraw ETH or tokens to a specified destination address with a regular transaction
+   * @param generationPrivateKey Receiver's generation private key
    * @param token Address of token to withdraw
-   * @param recipient Recipient address
+   * @param stealthAddress Stealth address funds were sent to
+   * @param destination Address to send funds to
```

Comment:
Nit: might reword to "Address where funds will be withdrawn to" to avoid confusion with "send" in previous param's description.

Reply:
Done in ba0e6cbc548a145aa90dc54534b6baef8f192ff5